include/nfl/poly.hpp

```diff
@@ -151,6 +151,15 @@ class poly {
   void ntt_pow_phi() { base.ntt_pow_phi(*this);}
   void invntt_pow_invphi() { base.invntt_pow_invphi(*this); }
 
+  /* manual serializers
+  */
+  inline void serialize(std::ostream& outputstream) {
+    outputstream.write(reinterpret_cast<char*>(_data), N * sizeof(T));
+  }
+  inline void deserialize(std::istream& inputstream) {
+    inputstream.read(reinterpret_cast<char*>(_data), N * sizeof(T));
+  }
```

Comment:
BTW, this kind of serialization is not portable in terms of « cross architecture portability ». It should be mentionned in the doc.